### PR TITLE
Use overlapped signatures of _winapi functions

### PR DESCRIPTION
**Pull request by ethanhs**
For https://github.com/python/mypy/pull/6148.

### CODE REVIEW

File: stdlib/3/_winapi.pyi [hunk 1]
```diff
@@ -45,7 +45,7 @@ WAIT_OBJECT_0: int
 WAIT_TIMEOUT: int
 
 def CloseHandle(handle: int) -> None: ...
-def ConnectNamedPipe(handle: int, overlapped: Union[int, bool] = ...) -> None: ...
+def ConnectNamedPipe(handle: int, overlapped: Union[int, bool] = ...) -> Overlapped: ...
```
Comment (srittau) [inline stdlib/3/_winapi.pyi]: This will always return `None` if `overlapped` is not provided or is False.

File: stdlib/3/_winapi.pyi [hunk 2]
```diff
@@ -63,13 +63,13 @@ def GetStdHandle(std_handle: int) -> int: ...
...
 def PeekNamedPipe(handle: int, size: int = ...) -> Union[Tuple[int, int], Tuple[bytes, int, int]]: ...
-def ReadFile(handle: int, size: int, overlapped: Union[int, bool] = ...) -> Tuple[bytes, int]: ...
+def ReadFile(handle: int, size: int, overlapped: Union[int, bool] = ...) -> Tuple[Overlapped, int]: ...
 def SetNamedPipeHandleState(named_pipe: int, mode: Optional[int], max_collection_count: Optional[int], collect_data_timeout: Optional[int]) -> None: ...
 def TerminateProcess(handle: int, exit_code: int) -> None: ...
 def WaitForMultipleObjects(handle_seq: Sequence[int], wait_flag: bool, milliseconds: int = ...) -> int: ...
 def WaitForSingleObject(handle: int, milliseconds: int) -> int: ...
 def WaitNamedPipe(name: str, timeout: int) -> None: ...
-def WriteFile(handle: int, buffer: bytes, overlapped: Union[int, bool] = ...): ...
+def WriteFile(handle: int, buffer: bytes, overlapped: Union[int, bool] = ...) -> Tuple[Overlapped, int]: ...
```
Comment (srittau) [inline stdlib/3/_winapi.pyi]: This will return a `Tuple[int, int]` if overlapped is `False`.

File: stdlib/3/_winapi.pyi [hunk 3]
```diff
@@ -63,13 +63,13 @@ def GetStdHandle(std_handle: int) -> int: ...
 def GetVersion() -> int: ...
 def OpenProcess(desired_access: int, inherit_handle: bool, process_id: int) -> int: ...
 def PeekNamedPipe(handle: int, size: int = ...) -> Union[Tuple[int, int], Tuple[bytes, int, int]]: ...
-def ReadFile(handle: int, size: int, overlapped: Union[int, bool] = ...) -> Tuple[bytes, int]: ...
+def ReadFile(handle: int, size: int, overlapped: Union[int, bool] = ...) -> Tuple[Overlapped, int]: ...
```
Comment (srittau) [inline stdlib/3/_winapi.pyi]: This will return `Tuple[bytes, int]` if overlapped is not provided or `False`. An `@overload` seems like the best solution.

**Comment by ethanhs** [comment]
@srittau I can provide an overload if overlapped is `None` but otherwise I'm not sure what you are suggesting? For example for `ConnectNamedPipe` the correct signature is something like:
```python
@overload
def ConnectNamedPipe(handle: int, overlapped: Literal[True, 1]) -> Overlapped: ...
@overload
def ConnectNamedPipe(handle: int, overlapped: Optional[Literal[False, 0]] = None) -> None: ...
@overload
def ConnectNamedPipe(handle: int, overlapped: Union[bool, int, None] = None) -> Optional[Overlapped]: ...
```

but we aren't at the point where I can start using Literal types in typeshed, correct?

**Comment by srittau** [comment]
Um, good point. I blame too much typeshed lately. Then we should just `Any` as return value for now (except for the `None`/not provided case).

**Comment by ethanhs** [comment]
I realized I was wrong, and `None` is not a valid argument to `overlapped`, so I just did an overload of where it is omitted. Hopefully this looks like what you wanted?

**Comment by ethanhs** [comment]
Indeed! Looking forward to being able to use them :)

**Pull request merged**